app/questionnaire/router.py

```diff
@@ -0,0 +1,52 @@
+class Router:
+
+    def __init__(self, schema, routing_path, completeness, current_location=None):
+        self._schema = schema
+        self._routing_path = routing_path
+        self._completeness = completeness
+        self._current_location = current_location
+
+    def can_access_location(self):
+        if self._is_valid_location() and not self._get_first_incomplete_block_in_section() and not self._get_first_incomplete_block_in_survey():
+            return True
+
+        return False
+
+    def get_next_location(self):
+        if not self._is_valid_location():
+            next_location = self._get_expected_location()
+            return next_location
+
+        return self._get_first_incomplete_block_in_section() or self._get_first_incomplete_block_in_survey() or self._get_expected_location()
+
+    def _is_valid_location(self):
```

Comment:
Since we have a couple of "location" concepts floating around (location that the user is trying to get to, location that we're _actually_ sending them to because we aren't allowing that location, location that we're redirecting them to because they just want to know where to go next...) I think it's worth changing this method name to something like `_is_current_location_valid`.